synapse/events/spamcheck.py

```diff
@@ -71,6 +71,11 @@
         Union[
             Literal["NOT_SPAM"],
             "synapse.api.errors.Codes",
+            # Highly experimental, not officially part of the spamchecker API, may
+            # disappear without warning depending on the results of ongoing
+            # experiments.
+            # Use this to return additional information as part of an error.
+            Tuple["synapse.api.errors.Codes", Dict],
```

Comment:
Similarly, I think we should use `JsonDict` here instead of `Dict` (sorry I missed that in https://github.com/matrix-org/synapse/pull/12846).